server/clustering.go

```diff
@@ -572,9 +602,75 @@ func (r *raftFSM) Apply(l *raft.Log) interface{} {
 		}
 		return nil
 	case spb.RaftOperation_DeleteChannel:
+		// Delete only if the channel exists and has the same ID.
+		if r.lookupChannel(op.Channel, op.ChannelID) == nil {
+			return nil
+		}
 		s.processDeleteChannel(op.Channel)
 		return nil
 	default:
 		panic(fmt.Sprintf("unknown op type %s", op.OpType))
 	}
 }
+
+// This returns a channel object only if it is found with the proper ID.
+func (r *raftFSM) lookupChannel(name string, id uint64) *channel {
+	s := r.server
+	cs := s.channels
+	cs.RLock()
+	defer cs.RUnlock()
+
+	c := cs.channels[name]
+	// Consider no ID (either in channel or from param) to be a match.
+	if c != nil && (id == 0 || c.id == 0 || c.id == id) {
```

Comment:
What does id == 0 signify?

Reply:
This is for existing channels that were created before this fix.

Reply:
Gotcha. Maybe add comment.